### PR TITLE
Small cleanup on protocol cleaning

### PR DESCRIPTION
**Pull request by jecisc**
- cleanUpProtocolsForClass: is always called with the organized class as parameter ==> Remove the parameter
- The name is not really explicit. What cleanup? I propose removeUnexistingSelectorsFromProtocols
- The method had no test which is now fixed 

Bonus: ClassOrganizationTest should use the actual organization of the class and not create a duplicated one to keep coherence in the tests.

### CODE REVIEW

File: src/Kernel/ClassOrganization.class.st [hunk 1]
```diff
@@ -232,6 +225,15 @@ ClassOrganization >> removeProtocolIfEmpty: aProtocol [
 	self notifyOfChangedProtocolNamesFrom: oldProtocolNames to: self protocolNames
 ]
 
+{ #category : #cleanup }
+ClassOrganization >> removeUnexistingSelectorsFromProtocols [
+	"For each protocol, remove the selectors that are not present in the class."
```
Comment (MarcusDenker) [inline src/Kernel/ClassOrganization.class.st]: "Unexisting" in english sounds strange... but it exists: "Adjective. unexisting (not comparable) Not in existence."

**Review by MarcusDenker — APPROVED** [review]

**Comment by MarcusDenker** [comment]
There are now conflicts

**Comment by jecisc** [comment]
I'll ask some native english speaker friends if they can think about a better name

**Comment by MarcusDenker** [comment]
the alternative is "nonexistent" https://www.merriam-webster.com/dictionary/unexisting

But I am fine as long as it has a dictionary entry

<blockquote><div><strong><a href="https://www.merriam-webster.com/dictionary/unexisting">Definition of UNEXISTING</a></strong></div><div>nonexistent… See the full definition</div></blockquote>

**Comment by jecisc** [comment]
Native English speaker says that nonexistent is a better word indeed. I'll resolve the conflict and rename

**Comment by jecisc** [comment]
I renamed to use nonexistent

**Comment by jecisc** [comment]
I forgot to rename the tests and I don't have the time now. I'll do that later

**Review by MarcusDenker — APPROVED** [review]

**Pull request merged**